src/Calendar.jsx

```diff
@@ -153,7 +157,7 @@ let Calendar = React.createClass({
     //if the value changes reset views to the new one
     if ( !dates.eq(val, dateOrNull(this.props.value), VIEW_UNIT[view]))
       this.setState({
-        currentDate: val ? new Date(val) : new Date()
+        currentDate: this.getDateOrCurrent(val)
```

Comment:
the Wrapped Component will inject onCurrentDateChange, but the Base one won't. Since the DatePicker uses the Plain ol Calendar the easiest way to handle that is to include currentDate/onCurrentDateChange here as well: https://github.com/jquense/react-widgets/blob/master/src/DateTimePicker.jsx#L464